### PR TITLE
Reload spaces re-work

### PR DESCRIPTION
**Pull request by SimonRichardson**
## Description of change

The following re-works the reload spaces by moving it up one level into
the environs level. This allows us to make better decisions about how to
reload the spaces.

The logic that defines how and what reload spaces should do, should not
be at the state (model), but instead, be controlled via the facade (controller).

This is a baby step move to allow all other locations to access reload
spaces.

Eventually, with more knowledge, we believe that reload spaces should
actually be on the environ and the results should then be sent to state.
At the moment it's a bit backward and that's a shame.

In reality, ReloadSpaces should return a list of NetworkPlanes (for a
better word). Those can then tell you if it's a space or a subnet,
sending each space to be saved to spaces collection and the subnets to
the subnet collection.

## QA steps

Test pass.

----

Bootstrap aws, create a subnet through the console.

```
juju reload-spaces
juju show-spaces
```

### CODE REVIEW

**Comment by SimonRichardson** [comment]
https://trello.com/c/t6XZSqac/2444-move-state-reloadspaces-to-the-api-which-then-calls-the-correct-state-methods

File: environs/spaces/spaces.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package spaces
```
Comment (manadart) [inline environs/spaces/spaces.go]: Can we just make this `space`? It irks me that there are packages named `spaces`, when we use the singular in almost all other cases. Even then is it reasonable to just add to `environs.networking`?

File: state/spacesdiscovery.go [hunk 2]
```diff
@@ -29,35 +26,6 @@ func (st *State) getModelSubnets() (set.Strings, error) {
 	return modelSubnetIds, nil
 }
 
-// ReloadSpaces loads spaces and subnets from provider specified by environ into state.
-// Currently it's an append-only operation, no spaces/subnets are deleted.
-func (st *State) ReloadSpaces(environ environs.BootstrapEnviron) error {
```
Comment (manadart) [inline state/spacesdiscovery.go]: This separate file is probably not required now. Just poke the methods into the subnets/spaces files.

`SaveProviderSpaces` and `SaveProviderSubnets` are better canonical names too, but feel free to ignore my pedantry there.

**Comment by SimonRichardson** [comment]
$$merge$$

**Pull request merged**